ci/test/offline

```diff
@@ -5,6 +5,32 @@ source ./common
 init_test_repo
 cd "$test_repo_path"
 
+# generate the workfow id
+cat <<EOF > id.py
+import os
+import hashlib
+identifier = str(os.getuid()) + "_main.workflow"
+id = str(hashlib.md5(identifier.encode()).hexdigest())
+print(id)
+EOF
+
+ID=$(python id.py 2>&1)
+
+if [ "$RUNTIME" == "vagrant" ]; then
+mkdir -p "$HOME/.cache/.popper/vagrant/$ID"
+cd "$HOME/.cache/.popper/vagrant/$ID"
+cat <<EOF > Vagrantfile
+Vagrant.configure("2") do |config|
+    config.vm.box = "ailispaw/barge"
+    config.vm.synced_folder "$HOME", "$HOME"
+    config.vm.synced_folder "/tmp/mypaper", "/tmp/mypaper"
+end
+EOF
+vagrant up
+export DOCKER_HOST=tcp://0.0.0.0:2375
```

Comment:
isn't this instructing `docker-py` to connect to the host's docker instead of the one inside the VM?

Reply:
@ivotron I double-checked this in a completely new ubuntu server. I installed only `vagrant` and `virtual-box` and not `docker`. The workflow ran perfectly fine. 

```
(venv) root@ubuntu-s-4vcpu-8gb-blr1-01:~/.cache/.popper/vagrant/14f0e17d3614e6ffed27839b45f1718b# docker

Command 'docker' not found, but can be installed with:

snap install docker     # version 18.06.1-ce, or
apt  install docker.io

See 'snap info docker' for additional versions.

(venv) root@ubuntu-s-4vcpu-8gb-blr1-01:~/.cache/.popper/vagrant/14f0e17d3614e6ffed27839b45f1718b# vagrant ssh
Welcome to Barge 2.13.0, Docker version 1.10.3, build 20f81dd
[bargee@barge ~]$ docker ps -a
CONTAINER ID        IMAGE                                           COMMAND                  CREATED              STATUS                          PORTS               NAMES
fba885632c87        popperized/bin:master                           "/entrypoint.sh env"     About a minute ago   Exited (0) About a minute ago                       popper_deploy_14f0e17d3614e6ffed27839b45f1718b
7ab650af3b4b        ef25486728ce                                    "/entrypoint.sh branc"   2 minutes ago        Exited (0) 2 minutes ago                            popper_branch-filter_14f0e17d3614e6ffed27839b45f1718b
81b135487ba7        popperized/github-actions-demo/jshint:9ec4d31   "jshint"                 2 minutes ago        Exited (0) 2 minutes ago                            popper_lint_14f0e17d3614e6ffed27839b45f1718b
050b15c8ec60        node:11.6.0                                     "npm test"               3 minutes ago        Exited (0) 3 minutes ago                            popper_test_14f0e17d3614e6ffed27839b45f1718b
dd2fd667db21        node:11.6.0                                     "npm install"            4 minutes ago        Exited (0) 3 minutes ago                            popper_install_14f0e17d3614e6ffed27839b45f1718b
[bargee@barge ~]$ docker images
REPOSITORY                              TAG                 IMAGE ID            CREATED              SIZE
popperized/bin                          master              768a0f81305c        About a minute ago   69.22 MB
<none>                                  <none>              ef25486728ce        2 minutes ago        70.34 MB
popperized/github-actions-demo/jshint   9ec4d31             706832236a60        2 minutes ago        905.2 MB
debian                                  stable-slim         eece114ab04d        4 days ago           69.21 MB
node                                    11.6.0              72ff1a4450d9        9 months ago         898 MB
[bargee@barge ~]$ 
```

As you can see `docker` is not installed on the `ubuntu` server but upon `ssh` into the VM, we can see the images and containers are created by `docker-py` inside the VM using the daemon running inside the VM.

Reply:
Also, `docker-daemon` on host by default listens on the socket `unix:///var/run/docker.sock` and here inside the VM it listens by default on `tcp://127.0.0.1:2375` along with unix socket.